schemas/public/onpremises-kfd-v1alpha2.json

```
@@ -518,33 +539,34 @@
           "items": {
             "type": "string"
           },
-          "description": "The tls cipher suites to use"
+          "description": "The TLS cipher suites to use for etcd, kubelet, and kubeadm static pods. Example:\n```yaml\ntlsCipherSuites:\n  - \"TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256\"\n  - \"TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256\"\n  - \"TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384\"\n  - \"TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384\"\n  - \"TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256\"\n  - \"TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256\"\n  - \"TLS_AES_128_GCM_SHA256\"\n  - \"TLS_AES_256_GCM_SHA384\"\n  - \"TLS_CHACHA20_POLY1305_SHA256\"\n```"
         },
         "configuration": {
           "type": "string",
-          "description": "The configuration to use"
+          "description": "etcd's encryption at rest configuration. Must be a string with the EncryptionConfiguration object in YAML. Example:\n\n```yaml\n\napiVersion: apiserver.config.k8s.io/v1\nkind: EncryptionConfiguration\nresources:\n  - resources:\n    - secrets\n    providers:\n    - aescbc:\n        keys:\n        - name: mykey\n          secret: base64_encoded_secret\n```\n"
         }
       }
     },
     "Spec.Kubernetes.Advanced.AirGap": {
       "type": "object",
+      "description": "Advanced configuration for air-gapped installations. Allows setting custom URLs where to download the binaries dependencies from and custom .deb and .rpm package repositories.",
       "additionalProperties": false,
       "properties": {
         "containerdDownloadUrl": {
           "type": "string",
-          "description": "The containerd download url"
+          "description": "URL where to download the containerd binary from."
         },
         "runcDownloadUrl": {
           "type": "string",
-          "description": "The runc download url"
+          "description": "URL where to download the runc binary from."
         },
         "runcChecksum": {
           "type": "string",
-          "description": "The runc checksum"
+          "description": "Checksum for the runc binary."
         },
         "etcdDownloadUrl": {
           "type": "string",
-          "description": "The etcd download url"
+          "description": "URL where to download the etcd binary from."
```

Is this the binary or the tar.gz ?

---

@alessiodionisi please confirm

---

Maybe is the binary folder, the default is `https://storage.googleapis.com/etcd`.

---

Yes, it's the folder of the archives:

```
- name: Getting etcd release
  unarchive:
    src: "{{ etcd_download_url }}/{{ etcd_version }}/etcd-{{ etcd_version }}-linux-{{ host_architecture }}.tar.gz"
    dest: /tmp
    remote_src: yes
```

---

```suggestion
          "description": "URL to the path where the etcd `tar.gz`s are available. etcd will be downloaded from `<etcdDownloadUrl>/<etcd_version>/etcd-<etcd_version>-linux-<host_architecture>.tar.gz`"
```

@alessiodionisi @nutellinoit ok?

---

LGTM